### PR TITLE
Make BindAllArchetypes resilient against DoArchetypeBindingForType fa…  …

### PR DESCRIPTION
**Pull request by adrian-prantl**
…iling.

rdar://problem/58994724

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

File: lldb/source/Symbol/SwiftASTContext.cpp [hunk 1]
```diff
@@ -5431,7 +5431,9 @@ static CompilerType BindAllArchetypes(CompilerType type,
     return type;
   ExecutionContext exe_ctx;
   exe_scope->CalculateExecutionContext(exe_ctx);
-  return runtime->DoArchetypeBindingForType(*frame, type);
+  if (auto bound = runtime->DoArchetypeBindingForType(*frame, type))
+    return bound;
+  return type;
```
Comment (dcci) [inline lldb/source/Symbol/SwiftASTContext.cpp:5436]: I thought we should really never fail unless we have a type parameter we're trying to bind.
( i.e. https://github.com/apple/llvm-project/pull/549 )

Comment (dcci) [reply lldb/source/Symbol/SwiftASTContext.cpp:5436]: I take this back.

**Review by dcci — APPROVED** [review]
LGTM.

**Review by fredriss — APPROVED** [review]

**Pull request merged**